### PR TITLE
Solve: 1 lazy backend result implementation

### PR DESCRIPTION
**Pull request by gjeusel**

### CODE REVIEW

**Comment by gjeusel** [comment]
Hello @epandurski !
I went for an easy solution, avoiding to infer the Result Backend instance based on configuration.
It will still be up to the user to add this middleware in their respective `app_factory`.

Tell me what you think of it ! 😉 

(I'll also need a new tag if possible !)

File: flask_melodramatiq/lazy_broker.py [hunk 1]
```diff
@@ -141,10 +146,21 @@ def init_app(self, app):
             self.__configuration = configuration
             options = configuration.copy()
             self.__class__ = options.pop('class')
+
+            # Instanciate dramatiq Broker
             broker = self._dramatiq_broker_factory(**options)
+
+            # Pull out empty backend from middleware
+            if self.__empty_backend in broker.middleware:
+                broker.middleware.remove(self.__empty_backend)
```
Comment (epandurski) [inline flask_melodramatiq/lazy_broker.py]: You instantiate the broker with `self.__empty_backend` (by passing **options), and then remove it. This seems potentially dangerous to me. Dramatiq does not guarantee that this is safe. It seems more robust to not include it in `options` at all. For example, in place of:

```Python
self.__options['middleware'] = self.__stub.middleware
```
you can have something like this:

```Python
self.__options['middleware'] = [m for m in self.__stub.middleware if m is not self.__empty_backend]
```
That is, if I am not missing something.

**Comment by epandurski** [comment]
What will be a good way to configure the backend in the app factory, in this case? (I have not used dramatiq result backends so far).

**Comment by epandurski** [comment]
Hi @gjeusel 

Probably a stupid question, but what do you mean by needing a tag? A github tag, or a new release in PyPi?

File: flask_melodramatiq/lazy_broker.py (hunk 1 above)
Comment (gjeusel) [reply flask_melodramatiq/lazy_broker.py]: Much better indeed !

**Comment by gjeusel** [comment]
Then my dramatiq_factory called by my flask app_factory looks like the following:

```python
import dramatiq
from dramatiq.results import Results
from dramatiq.results.backends.redis import RedisBackend
from dramatiq.results.backends.stub import StubBackend

def dramatiq_factory(app, testing):
    melodrama_broker = dramatiq.get_broker()

    if testing:
        backend = StubBackend()
    else:
        backend = RedisBackend(url=app.config["DRAMATIQ_BROKER_URL"])

    melodrama_broker.add_middleware(Results(backend))
    melodrama_broker.init_app(app)
    return melodrama_broker
```

**Comment by gjeusel** [comment]
Yep, I'll need a new PyPi release plz ! (That's what I meant by new tag)
Thanks in advance !! 😄

**Comment by epandurski** [comment]
Done.

https://pypi.org/project/Flask-Melodramatiq/

**Pull request merged**